src/cron/updateCurrentStreak.ts

```diff
@@ -16,13 +16,16 @@ const cron: Cron = {
       await con.transaction(async (entityManager): Promise<void> => {
         const usersPastStreakTime = await entityManager
           .createQueryBuilder()
-          .select(
-            `us.*, (date_trunc('day', us."lastViewAt" at time zone COALESCE(u.timezone, 'utc'))::date) AS "lastViewAtTz", u.timezone`,
+          .select(`us.*`)
+          .addSelect(
+            `date_trunc('day', us."lastViewAt" at time zone COALESCE(u.timezone, 'utc'))::date`,
+            'lastViewAtTz',
           )
+          .addSelect('u.timezone', 'timezone')
           .addSelect('us.currentStreak', 'current')
           .addSelect('u."weekStart"', 'weekStart')
           .addSelect(
-            `(date_trunc('day', usa."lastRecoverAt" at time zone COALESCE(u.timezone, 'utc'))::date) - interval '1 day'`,
+            `(date_trunc('day', usa."lastRecoverAt"::timestamptz at time zone COALESCE(u.timezone, 'utc'))::date) - interval '1 day'`,
```

Comment:
Didn't we discuss not doing the timestamptz columns for lots of reasons?
Not sure if we ended up not doing it, but they such a pain...

Reply:
Yes, this column is not `timestamptz`. The problem occurs when you apply the time zone to a column similar to this (just `timestamp`).

Using `at time zone` in a non-timezone column messes up the values rather than adjusting to the right timezone. For us to use the `at time zone` statement, the column must be converted to `timestamptz` first even when the column is just a pure `timestamp`. Some nuances working with timezone in postgresql.

Reply:
Potentially we can extract `date_trunc(...)` and everything else in a function maybe so the query is a bit more readable. Does not need to happen in this PR just a note it is starting to be really hard to follow 😆

Reply:
Also, we are using `AT TIME ZONE` in digest and it correctly transforms there from what I see. But there we convert from `NOW()` so not sure if that is the difference.

Reply:
I guess because `NOW()` includes the timezone offset
<img width="360" alt="image" src="https://github.com/user-attachments/assets/6eda2d53-207e-4bac-a0dc-5f5c8397239f">

Reply:
Yeah, using `now` works as intended since it includes the timezone information. Basically a `timestamptz` type is the value returned from the function.

Reply:
> Potentially we can extract date_trunc(...) and everything else in a function maybe so the query is a bit more readable. Does not need to happen in this PR just a note it is starting to be really hard to follow 😆

Totally agree, especially have it in many places. Ticker created here: https://github.com/dailydotdev/daily-api/pull/2452#discussion_r1849688689